File: examples/python/polite_responder.py
from trainloop_llm_logging import collect, trainloop_tag

collect(flush_immediately=True)

from ai_request import make_ai_request

if __name__ == "__main__":
    complaint = "Your package arrived two weeks late and the box was damaged!"
    prompt = f"""You are a customer-support agent.
Reply to the following customer in ≤120 words.
Requirements:
1. Begin with a sincere apology
2. Acknowledge the specific problem
3. Offer a concrete next step or compensation
Customer: "{complaint}"
"""
    headers = trainloop_tag("polite-responder")
    print(make_ai_request(prompt, extra_headers=headers))

File: examples/typescript/politeResponder.ts
// Notice that trainloop logging should be setup via --require flag in package.json script
import { trainloopTag, collect } from 'trainloop-llm-logging';
collect(true);

import { makeAiRequest } from './aiRequest';

// Example usage
async function main() {
    const complaint = "Your package arrived two weeks late and the box was damaged!";
    const promptText = `You are a customer-support agent.
Reply to the following customer in ≤120 words.
Requirements:
1. Begin with a sincere apology
2. Acknowledge the specific problem
3. Offer a concrete next step or compensation
Customer: "${complaint}"`;

    // Tag this request for the polite responder evaluation suite
    const headers = trainloopTag("polite-responder");
    const response = await makeAiRequest(promptText, "gpt-4o-mini", 500, headers);
    
    if (response) {
        console.log("AI Response:", response);
    }
    process.exit(0);
}

if (require.main === module) {
    main().catch(console.error);
} 